src/coreclr/jit/importercalls.cpp

```diff
@@ -3093,6 +3094,29 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Type_get_TypeHandle:
+            {
+                assert(IsTargetAbi(CORINFO_NATIVEAOT_ABI));
```

Comment:
Can we do for all runtimes? This does not look CoreCLR specific.

Reply:
> Can we do this for all runtimes? This does not look NativeAOT specific.

But how'd it work for CoreCLR where `RuntimeTypeHandle` is a wrapper around `RuntimeType` object? It only works for NativeAOT where `RuntimeTypeHandle` is a wrapper around class handle, see https://github.com/dotnet/runtime/pull/85804#issuecomment-1536612618